chapters/11-version-control-workflow.Rmd

```diff
@@ -1,181 +1,600 @@
-    and runs `git pull upstream main`,
-    then merges any changes or conflicts.
-7.  Fuchsia pushes from the `main` branch of their desktop repository
-    to the `main` branch of their GitHub repository.
-8.  Chartreuse merges Fuchsia's pull request.
-9.  Chartreuse runs `git pull origin main` on the desktop.
+It outputs quite a bit of text that you can use when you encounter problems and
+you aren't sure how to diagnose the problem. The second helper function is
+`gh_token_help()`
 
-### Branches and forks
+```{r cs-gh-token-help}
+gh_token_help()
+```
 
-- review branching from solo git chapter
-- introduce forks and relate to branching
+This output tells you if you have a GitHub token already and what to do
+to either create a new one (using `create_github_token()`) or how to access
+an already existing one `gitcreds::gitcreds_set()`. 
+For now, we will create a new one just in case one hasn't been created yet.
 
-- How to use each effectively
-    - How branches can help both solo and as a team
-    - When forks are good vs being contributor on a repo (write access)
-    
-## Collaborating with GitHub
+```{r cs-create-token, eval=FALSE}
+create_github_token()
+```
 
-- What are PRs, how to use them (either non-fork or fork)
-    - Keeping things targeted (e.g. a PR should conceptually
-    fix or add similarly related things)
-    
-### Exercise: Reading and comprehension task?
+This function sends us to the GitHub "Generate new token" webpage with all the
+necessary settings checked. All you need to do is click the green button at the
+bottom called "Generate token" and you'll have a very long string generated
+for you. **Save this token in your password manager**. This is the token you 
+will use every time you open up RStudio and interact with GitHub through R.
+You do **not** need to create a new token for each R project or package you
+make, you only need to create one after your current token expires 
+(typically every couple of months), if you've forgotten the token or lost it,
+or if you've changed to a new computer.
+In order for R recognize the token, you need to run the below code in the Console:
+
+```{r cs-gitcreds-set, eval=FALSE}
+gitcreds::gitcreds_set()
+```
 
-### Exercise: Do these situations warrant a branch or fork?
+And then copy and paste your token into the prompt in the Console. The gitcreds
+package typically saves this token for the day (it caches the token), but after
+restarting you will need to run the action again. 
 
-Concept and thought, not code based.
+### Exercises: Consider your own computer security practices
 
-1. 
-2. 
-3. 
+TODO: Thoughts on these ones? Might be a bit provocative :P
 
-### Exercise: Consider some situations that would benefit from using PRs
+Take some time to think about these questions and self-reflect on them.
 
-Concept and thought, not code based.
+1. Do you use a password manager? Why or why not? If not, what is it that is
+holding you back from starting to use one?
```

Comment:
This is obviously very important, but is it outside the scope of the book? I suppose you have told them earlier to save their token in a password manager... not sure where I stand on this one.

Reply:
Hmmm, its a good question... My thought is that considering we are talking about collaboration, and collaboration on an open and public repository, security is an important consideration. I don't know how to disentangle this... I'll add a TODO item here to discuss this more eventually.